docs/contributors/patching/maintain-ubuntu-desktop-software.md

```diff
@@ -0,0 +1,1110 @@
+
+
+## Team members, contributors and permissions
+
+If you're a community contributor and not a member of the Ubuntu Desktop Team, you have to send your contributions through a merge request:
+
+1. Fork the Git repository on Salsa.
+1. Follow this guide and make the changes in your fork.
+1. Open a Salsa merge request from your fork to the original repository.
+
+In some projects, no Ubuntu branch has been created in a long time. You might have to ask the Desktop Team to create the new branch for you. You can contact them on Matrix at {matrix}`desktop-dev` or [on the Canonical Mattermost](https://chat.canonical.com/canonical/channels/desktop).
+
+
+## Local changes
+
+With the suggested Git configuration, any non-committed changes (file modifications, additions or removal) halt the build because they won't be included. This is a safety reminder.
+
+What you can do when you have such changes:
+
+* Ignore them. The resulting package and build won't include those uncommitted changes. The `../build-area/<your-package>` directory will match the last commit.
+
+    To do this, add the `--git-ignore-new` Git option
+
+* Force using the current directory with your local modifications instead of the `build-area` directory, and include any local modifications to the package despite the `ignore-new` option.
+
+    For this, add the `--git-ignore-new --git-export=INDEX` options to `gbp`.
+
+
+## Refresh branches
+
+Use the following command to refresh the current `ubuntu/<release>` branch and the corresponding `upstream/<release>` branch. This applies only to the branches referenced in the `debian/gbp.conf` file on your current checkout. Also refresh the `pristine-tar` branch. 
+
+* For example, on the `ubuntu/latest` branch:
+
+    ```{terminal}
+    :copy:
+    :host:
+    :dir: gnome-control-center
+    :user:
+    :input: gbp pull
+
+    gbp:info: Fetching from default remote for each branch
+    gbp:info: Branch 'upstream/latest' is already up to date.
+    gbp:info: Branch 'ubuntu/latest' is already up to date.
+    gbp:info: Updating 'pristine-tar': 2bed57f20816..e59ab7422a13
+    ```
+
+    The `pristine-tar` branch has been updated as referenced in `debian/gbp.conf`.
+
+* If you have a separate `ubuntu/noble` branch, with its own `upstream/46.x` branch:
+
+    ```{terminal}
+    :copy:
+    :host:
+    :dir: gnome-control-center
+    :user:
+    :input: git checkout ubuntu/noble
+
+    :input: gbp pull
+
+    gbp:info: Fetching from default remote for each branch
+    gbp:info: Branch 'ubuntu/noble' is already up to date.
+    gbp:info: Branch 'upstream/46.x' is already up to date.
+    gbp:info: Branch 'pristine-tar' is already up to date.
+    ```
+
+The `gbp pull` command, contrary to `git pull`, is a way to avoid checking out each branch and pulling them one by one. With `git pull`, you need to have the branch as the current checkout for potential conflicts when pulling.
```

Comment:
As the `git pull` command isn't recommended, the sentence seems extra.
```suggestion
The `gbp pull` command, contrary to `git pull`, is a way to avoid checking out each branch and pulling them one by one.
```